### PR TITLE
Add intersectional type support under roblox flag

### LINKED ISSUE #62: Support intersectional types for Roblox

### PR DESCRIPTION
**Pull request by JohnnyMorganz**
`string & number`
Closes #62

### CODE REVIEW

File: full-moon/src/ast/parsers.rs [hunk 1]
```diff
@@ -1285,6 +1285,25 @@ cfg_if::cfg_if! {
                 state = new_state;
             }
 
+            // if let Ok((state, pipe)) = ParseSymbol(Symbol::Pipe).parse(state) {
```
Comment (Kampfkarren) [inline full-moon/src/ast/parsers.rs:1288]: Don't leave commented code in version control, just remove it.

File: full-moon/src/tokenizer.rs [hunk 2]
```diff
@@ -59,6 +59,8 @@ symbols!(
     Percent => "%",
     // TODO: This only is valid in Roblox
     Pipe => "|",
+    // TODO: This only is valid in Roblox
+    Ampersand => "&",
```
Comment (Kampfkarren) [inline full-moon/src/tokenizer.rs:63]: Alphabetical order with the rest of this group. You'll notice that this is sometimes out of order, but that's intentionally to make it so things like `.` don't trump `..`. This isn't one of those cases.

File: full-moon/src/ast/types.rs [hunk 3]
```diff
@@ -127,6 +127,20 @@ pub enum TypeInfo<'a> {
         #[cfg_attr(feature = "serde", serde(borrow))]
         right: Box<TypeInfo<'a>>,
     },
+
+    /// An intersection type: `string & number`, denoting both types.
+    #[display(fmt = "{}{}{}", "left", "ampersand", "right")]
+    Intersection {
```
Comment (Kampfkarren) [inline full-moon/src/ast/types.rs:133]: Alphabetical order.

File: full-moon/src/ast/parsers.rs (hunk 1 above)
Comment (JohnnyMorganz) [reply full-moon/src/ast/parsers.rs:1288]: I completely missed this, will remove

File: full-moon/src/tokenizer.rs [hunk 4]
```diff
@@ -15,6 +15,8 @@ use serde::{Deserialize, Serialize};
 use std::{borrow::Cow, cmp::Ordering, fmt, str::FromStr};
 
 symbols!(
+    // TODO: This only is valid in Roblox
+    Ampersand => "&",
```
Comment (Kampfkarren) [inline full-moon/src/tokenizer.rs:19]: By alphabetical order with the rest of the group, I mean with the rest of the single character symbols.

File: full-moon/src/tokenizer.rs [hunk 5]
```diff
@@ -37,6 +37,13 @@ symbols!(
     Until => "until",
     While => "while",
 
+    PlusEqual => "+=",
```
Comment (Kampfkarren) [inline full-moon/src/tokenizer.rs:40]: Needs the "TODO" comment on top of here too.

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Unreleased
 ### Added
 - Added support for `continue` under `roblox` feature flag
+- Added support for Compound Assignment (eg. `x += 1`) under the `roblox` feature flag
```
Comment (Kampfkarren) [inline CHANGELOG.md:10]:
```suggestion
 - Added support for compound assignments (eg. `x += 1`) under the `roblox` feature flag
```

**Comment by Kampfkarren** [comment]
Last changes and then I'll quickly make sure this all works on my end.

**Pull request merged**